uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java

```diff
@@ -77,6 +78,14 @@ public BrokerCompanyPortal(@NonNull final IAppInstaller appInstaller) {
         localApkFileName = COMPANY_PORTAL_APK;
     }
 
+    public BrokerCompanyPortal(final boolean isOldApk) {
```

Comment:
I think it is difficult for the readers of this code to figure out what would happen if we pass a true vs false to this constructor. It seems that we've got some boolean here that is governing how to pick the APK file name when doing install.

I think rather than having `oldApkFileName` and dynamically setting it based on a boolean, I think instead we should have clearly defined `localApkFileNameFreshInstall` and `localApkFileNameUpdate` - something like that and then you don't need a booleans in the constructor to figure these out

Reply:
The above feedback to all the places these patterns are used in this PR

Reply:
The reason I did not add localApkFileNameFreshInstall and localApkFileNameUpdate is because when we say update - we will actually be installing the latest apk (one that is picked from the latest pipeline) and so, when we run update scenarios, we first install an oldApk based on the testcase that is running. 

I could have kept static files like you mentioned. But in that case, I would end up installing localApkFileNameFreshInstall apks everytime for the first install - here it would become tricky what a fresh install means as we cannot differentiate between freshInstall in normal testcase and freshInstall in update testcase (in this case freshInstall means oldApk install)

To fix the ambiguity, I can add some comments to make it clearer.

Reply:
I think I kind of get what you're saying, but still not fully clear. Comments in the code would definitely help. Let's also sync on this offline next week so I can get a better understanding of the flow here.